[monitor-query] monitor query track 2

richardpark-msft opened the pull request:
Tracking issues here: [link](https://github.com/Azure/azure-sdk-for-js/issues/15296)

xirzec commented:
Left a bunch of comments on the APIView https://apiview.dev/Assemblies/Review/23c04df42f764b1c9b14ed9271a5191a

richardpark-msft commented:
@xirzec - I'll be going through the feedback but I figured we could start with this check as a baseline (if I can get it to properly build). 

I'm tracking additional work here: https://github.com/Azure/azure-sdk-for-js/issues/15296

xirzec commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,313 @@
+}
+
+// @public
+export interface LogQueryRequest {
```
this feels very strange. If we need to do something this lowlevel shouldn't we make a REST client instead?

richardpark-msft replied:
Agreed, I'll fix this up and probably just remove some of the lower level fields like `headers`, `method` and `path`.

richardpark-msft replied:
Fixed by not using it as the public model.

The newer one is simpler: https://github.com/richardpark-msft/azure-sdk-for-js/blob/edfc23f2f3c6854bbf254e3021c46cf593fa8638/sdk/monitor/monitor-query/src/models/logsModels.ts#L44

xirzec commented on sdk/monitor/monitor-query/src/internal/util.ts:
```diff
@@ -0,0 +1,35 @@
+export function formatPreferHeader(
+  args: Pick<QueryLogsOptions, "serverTimeoutInSeconds" | "includeQueryStatistics"> | undefined
+): { Prefer: string } | undefined {
+  if (args == null) {
```
feels a little weird since the input is `| undefined` and not `| null` and since it's an object type you could just use a falsy check, right?

richardpark-msft replied:
Added a falsy check.

xirzec commented on sdk/monitor/monitor-query/src/logger.ts:
```diff
@@ -0,0 +1,11 @@
+ *
+ * @internal
+ */
+export const logger = createClientLogger("template");
```
missing real name

richardpark-msft replied:
Fixed!

xirzec commented on sdk/monitor/monitor-query/src/logsClient.ts:
```diff
@@ -0,0 +1,63 @@
+    // host.
+    this._logAnalytics = new AzureLogAnalytics(tokenCredential, {
+      $host: options?.host,
+      credentialScopes: ["https://api.loganalytics.io/.default"]
```
do we really want to hardcode this?

richardpark-msft replied:
I moved it into a constant. I can investigate this some more, but I believe this is correct.

richardpark-msft replied:
I've opened an issue to track this:
https://github.com/Azure/azure-sdk-for-js/issues/15296

xirzec commented on sdk/monitor/monitor-query/src/logsClient.ts:
```diff
@@ -0,0 +1,63 @@
+import { QueryLogsBatchOptions, QueryLogsOptions, QueryLogsResult } from "./models/logsModels";
+import { formatPreferHeader } from "./internal/util";
+
+export interface LogsClientOptions {
```
not extending the base client options? (which for v1 is PipelineOptions)

richardpark-msft replied:
I can do that. I didn't intentionally target the old v1 so I might do a regen after this (and still extend the right options).

richardpark-msft replied:
Done. I also added in options for MetricsClient and did the same there.

xirzec commented on sdk/monitor/monitor-query/src/metricsClient.ts:
```diff
@@ -0,0 +1,60 @@
+  private _namespacesClient: GeneratedMetricsNamespacesClient;
+
+  constructor(tokenCredential: TokenCredential) {
+    this._metricsClient = new GeneratedMetricsClient(tokenCredential, "2018-01-01", {});
```
can we pull the api versions out into constants?

richardpark-msft replied:
Done. Using the very interestingly named auto-generated ones.

xirzec commented on sdk/monitor/monitor-query/src/metricsClient.ts:
```diff
@@ -0,0 +1,60 @@
+} from "./models/metricsModels";
+
+export class MetricsClient {
+  private _metricsClient: GeneratedMetricsClient;
+  private _definitionsClient: GeneratedMetricsDefinitionsClient;
+  private _namespacesClient: GeneratedMetricsNamespacesClient;
```
why are these three generated clients anyway? Seems silly for a single operation apiece

richardpark-msft replied:
I'm inexperienced in this - they're separate swaggers today. I'm not sure if there's a way to say "merge all of these swaggers into a single generated client" if that's what you're suggesting.

xirzec replied:
Seems like maybe a good autorest feature request? /cc @sarangan12 @joheredi

richardpark-msft replied:
To add to this - there are some duplicate contracts between them. So there's some possible savings on code generation there (perhaps not since it's still multiple endpoints).

xirzec commented on sdk/monitor/monitor-query/swagger/log_query_swagger.json:
```diff
@@ -0,0 +1,1438 @@
+{
```
feels a little strange having to check this in here

richardpark-msft replied:
Agreed. This is temporary - moving this stuff into the azure-rest repo is in process.

richardpark-msft replied:
Moved (referencing azure-rest-specs-api but using @srnagar's branch)

xirzec commented on sdk/monitor/monitor-query/test.env:
```diff
@@ -0,0 +1,26 @@
+# Used in most samples. Retrieve these values from an instance in the Azure
+# Portal.
+
+APPCONFIG_ENDPOINT="https://<resource name>.azconfig.io",
```
appconfig?

richardpark-msft replied:
Fixed.

xirzec commented on sdk/monitor/monitor-query/test/public/logsClient.spec.ts:
```diff
@@ -0,0 +1,139 @@
+        return result;
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, args.secondsBetweenQueries * 1000));
```
Seems like these tests are going to be slow, and we shouldn't need the timeout when doing playback, right?

richardpark-msft replied:
Very true. I think there's a way for me to "not record" pieces so what I'll probably do is skip recording when doing these calls, and only turn it back on in each of the underlying steps.

The "setup" part is not critical for recording anyways.

richardpark-msft replied:
There is a way to fix this properly when I use recordings. I'm having "unrelated to this" issues so this first PR is going to go with recordings shut off.

xirzec commented on sdk/monitor/monitor-query/tests.yml:
```diff
@@ -0,0 +1,11 @@
+stages:
+  - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
+    parameters:
+      PackageName: "@azure/template"
```
name

richardpark-msft replied:
Fixed

xirzec commented on sdk/monitor/monitor-query/tsconfig.json:
```diff
@@ -0,0 +1,11 @@
+    "outDir": "./dist-esm",
+    "declarationDir": "./types",
+    "paths": {
+      "@azure/template": ["./src/index"]
```
👀

richardpark-msft replied:
Also fixed (and became more obvious once I got my samples into samples-dev)

xirzec reviewed (changes requested):
This doesn't look quite ready to check-in, even from a "we just filled out the template standpoint."

In general I'm fine with "let's check in the template overhead first", but when doing so, please **don't** actually create the clients or any real test collateral, since that content actually makes it harder to notice when the template hasn't been customized completely.

Also, I'm uncomfortable with "don't review these particular files that I'm still checking in now" because in reality we're **never** going to review them holistically once they've been checked in. We're going to look at diffs and reviewers aren't going to notice anything off in non-changed parts of the file.

richardpark-msft commented on sdk/monitor/monitor-query/README-EXAMPLE.md:
```diff
@@ -0,0 +1,188 @@
+# App Configuration client library for JavaScript
```
Done!

richardpark-msft commented on sdk/monitor/monitor-query/README.md:
```diff
@@ -0,0 +1,102 @@
+# Azure Template client library for JavaScript
```
To stem my shame I instead filled this out maximally (ie, I believe it's actually fully accurate).

richardpark-msft commented on sdk/monitor/monitor-query/package.json:
```diff
@@ -0,0 +1,149 @@
+  "files": [
+    "dist/",
+    "dist-esm/src/",
+    "types/template.d.ts",
```
Fixed

richardpark-msft commented on sdk/monitor/monitor-query/review/template.api.md:
```diff
@@ -0,0 +1,44 @@
+## API Report File for "@azure/template"
```
Done

richardpark-msft commented on sdk/monitor/monitor-query/samples/v1/typescript/src/logsQueryBatchSample.ts:
```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
```
Also done!

richardpark-msft commented on sdk/monitor/monitor-query/samples/v1/typescript/src/logsQueryBatchSample.ts:
```diff
@@ -0,0 +1,79 @@
+  const kqlQuery = "AppEvents | project TimeGenerated, OperationName, AppRoleInstance | limit 1";
+
+  // Use our query, and get results for the last day.
+  const batchRequest: BatchRequest = {};
```
Removed!

richardpark-msft commented on sdk/monitor/monitor-query/samples/v1/typescript/src/logsQueryBatchSample.ts:
```diff
@@ -0,0 +1,79 @@
+  const logsClient = new LogsClient(tokenCredential);
+
+  if (!monitorWorkspaceId) {
+    throw new Error("MONITOR_WORKSPACE_ID must be set in the environment for this sample");
```
Fixed.

richardpark-msft commented on sdk/monitor/monitor-query/package.json:
```diff
@@ -0,0 +1,149 @@
+  "prettier": "@azure/eslint-plugin-azure-sdk/prettier.json",
+  "dependencies": {
+    "@azure/core-auth": "^1.3.0",
+    "@azure/core-http": "^1.2.0",
```
Talked with @ramya-rao-a - looks like @sarangan12  is working on getting autorest-core-v2 generation in at the moment. We're not shipping this library at the moment, but the first checkin is just going in with the mainline autorest generation (so still using v1).

richardpark-msft commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,313 @@
+}
+
+// @public
+export const last24Hours = "P1D";
```
Went with a constant object: `CommonDurations`

richardpark-msft commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,313 @@
+    message?: string;
+}
+
+// @public (undocumented)
```
I have a couple left - there is overlap with the swagger definitions so I'll need to figure out a way to unify the types prior to export.

richardpark-msft replied:
(these aren't going to disappear. I'm resolving this and just adding it to my issue tracking some of this follow-on work.

https://github.com/Azure/azure-sdk-for-js/issues/15296

richardpark-msft commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,313 @@
+export interface MetricNamespace {
+    id?: string;
+    name?: string;
+    // Warning: (ae-forgotten-export) The symbol "MetricNamespaceName" needs to be exported by the entry point index.d.ts
```
Fixed.

richardpark-msft commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,313 @@
+    // Warning: (ae-forgotten-export) The symbol "MetricDefinitionsListResponse" needs to be exported by the entry point index.d.ts
+    //
+    // (undocumented)
+    getMetricDefinitions(resourceUri: string, options?: GetMetricDefinitionsOptions): Promise<MetricDefinitionsListResponse>;
```
I don't believe the API has pageability built-in:

@srnagar, can you confirm? I believe this came up during the architecture discussion.

srnagar replied:
Yes, the service doesn't support pagination.

richardpark-msft commented on sdk/monitor/monitor-query/samples/v1/typescript/src/logsQueryBatchSample.ts:
```diff
@@ -0,0 +1,79 @@
+  for (const response of result.responses) {
+    console.log(`Results for query with id: ${response.id}`);
+
+    if (response.body?.errors) {
```
Null coallescing not okay?

richardpark-msft commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,313 @@
+    // (undocumented)
+    queryLogs(workspaceId: string, query: string, options?: QueryLogsOptions): Promise<QueryLogsResult>;
+    // (undocumented)
+    queryLogsBatch(body: BatchRequest, options?: QueryLogsBatchOptions): Promise<QueryBatchResponse>;
```
That could be really nice. Java has a nice type and it lets them be a bit more helpful when figuring out which fields make sense to fill out commonly (like workspaceId, rather than workspaces or workspaceIds), and auto-assign an ID, etc.. I'll take a look at data tables.

richardpark-msft replied:
I cleaned this up a bit. I can look at data-tables but the end result the user has to fill out doesn't end up being too bad (we still auto-generated the IDs and take care of ordering, etc..). 

The only "bad" thing is that there are some optional arrays that can be added onto for additional workspace Ids, etc..

richardpark-msft commented on sdk/monitor/monitor-query/package.json:
```diff
@@ -0,0 +1,149 @@
+    "typedoc": "0.15.2",
+    "typescript": "~4.2.0",
+    "util": "^0.12.1",
+    "@opentelemetry/tracing": "^0.18.2"
```
Done

richardpark-msft commented on sdk/monitor/monitor-query/package.json:
```diff
@@ -0,0 +1,149 @@
+    "check-format": "prettier --list-different --config ../../../.prettierrc.json --ignore-path ../../../.prettierignore \"src/**/*.ts\" \"test/**/*.ts\" \"*.{js,json}\"",
+    "clean": "rimraf dist dist-* temp types *.tgz *.log coverage coverage-browser",
+    "docs": "typedoc --excludePrivate --excludeNotExported --excludeExternals --stripInternal --mode file --out ./dist/docs ./src",
+    "execute:samples": "dev-tool samples run samples-dev",
```
Removed!

richardpark-msft commented:
/azp run prepare-pipelines

richardpark-msft commented:
/azp run js - monitor-query - tests

richardpark-msft commented:
@xirzec - I think I've addressed all the comments that I could see (in both the API view and here). The ones I've left unresolved are ones where I'm not sure and want to confirm that you're okay with my decisions.

Let me know if you want me to do anything more for this first release.

richardpark-msft commented:
Tacking on - I have now passed our (limited) live test run as well for the newly created monitor-query pipeline!

chradek commented on sdk/monitor/monitor-query/README.md:
```diff
@@ -0,0 +1,174 @@
+### Prerequisites
+
+- You must have an [Azure Subscription](https://azure.microsoft.com) and an [Azure Monitor][azure_monitor_product_documentation] resource to use this package.
+- Node.js version 10.x.x or higher
```
Nice.

richardpark-msft replied:
It did feel pretty good to write that. :)

chradek commented on sdk/monitor/monitor-query/README.md:
````diff
@@ -0,0 +1,174 @@
+Using [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/README.md#defaultazurecredential)
+
+```javascript
+const azureIdentity = require("@azure/identity");
+const monitorQuery = require("@azure/monitor-query");
+
+const credential = new azureIdentity.DefaultAzureCredential();
+
+const logsClient = new monitorQuery.LogClient(credential);
+// or
+const metricsClient = new monitorQuery.MetricsClient(credential);
````
Subjective:
What do you think of changing the code to use object destructuring so that it doesn't look like you're instantiating classes hanging off a namespace?

```suggestion
const { DefaultAzureCredential } = require("@azure/identity");
const { LogClient, MetricsClient } = require("@azure/monitor-query");

const credential = new DefaultAzureCredential();

const logsClient = new LogClient(credential);
// or
const metricsClient = new MetricsClient(credential);
```

richardpark-msft replied:
I prefer that. I'm not quite sure why I didn't do that this time around - will change.

richardpark-msft replied:
Done.

chradek commented on sdk/monitor/monitor-query/package.json:
```diff
@@ -0,0 +1,150 @@
+  ],
+  "repository": "github:Azure/azure-sdk-for-js",
+  "engines": {
+    "node": ">=8.0.0"
```
Match the README?
```suggestion
    "node": ">=10.0.0"
```

richardpark-msft replied:
(undone, for now). We haven't changed up the eslint rule yet and I don't want to add in an exclusion until we're ready to do it more broadly.

chradek commented on sdk/monitor/monitor-query/src/models/constants.ts:
```diff
@@ -0,0 +1,30 @@
+  last30Minutes: "PT30M",
+  /** Alias for ISO8601 value 'PT5M' */
+  last5Minutes: "PT5M"
+};
```
These constants won't change will they? Right now the type info makes it look like all these values are just strings, but you can have them be string literals if you add `as const` to the end of your assignment:

```typescript
const CommonDurations = {
/* fields */
} as const;
```

richardpark-msft replied:
Oh, interesting. I will make that fix, didn't know about that...but now I remember @witemple-msft doing something similar in dev-tool.

richardpark-msft replied:
Done

chradek commented on sdk/monitor/monitor-query/src/models/publicMetricsModels.ts:
```diff
@@ -0,0 +1,131 @@
+
+/** A time series result type. The discriminator value is always TimeSeries in this case. */
+export interface TimeSeriesElement {
+  // track 2 version of
```
I feel left in suspense.

richardpark-msft replied:
Fixed

chradek commented on sdk/monitor/monitor-query/src/models/publicMetricsModels.ts:
```diff
@@ -0,0 +1,131 @@
+  // track 2 version of MetricNamespacesListResponse
+
+  /** The metric namespaces. */
+  namespaces: MetricNamespace[] | undefined;
```
Is this field really always present, or should it be an optional field?

richardpark-msft replied:
Fixed - it should always be there.

chradek reviewed (approved):
No concerns from me. The API View isn't approved yet, I'd like to hear more about workspace vs workspaceId vs workspaces, but I'm ok if that gets punted to another PR.

xirzec commented on sdk/monitor/monitor-query/README.md:
```diff
@@ -0,0 +1,174 @@
+async function run() {
+  const kustoQuery = "AppEvents | limit 1";
+  const result = logsClient.queryLogs(azureLogAnalyticsWorkspaceId, kustoQuery);
+  const tablesFromResult: Table[] | undefined = result.tables;
```
this is a js code snippet but we have a TS type annotation in it :)

richardpark-msft replied:
Fixed...:|

xirzec commented on sdk/monitor/monitor-query/README.md:
```diff
@@ -0,0 +1,174 @@
+
+The following samples show you the various ways you can query your Log Analytics workspace:
+
+- [`logsQuery.ts`]<!--[samples_logsquery_ts]--> - Query logs in a Monitor workspace
```
should these be working links?

richardpark-msft replied:
Not yet (since we haven't checked in a first rev of this code into main). 

I'll basically do a commit right after this one to turn all of these links on.

xirzec commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,293 @@
+
+// @public
+export interface BatchQuery {
+    azureResourceIds?: string[];
```
whenever we have an array property, it's generally preferred that we initialize to the empty array if possible - is that possible for any properties on this type?

richardpark-msft replied:
This is a request type (not a response). Are you saying the caller should pass in empty arrays rather than just omitting the field (which is the intent here)

xirzec replied:
nah, I thought it was an output. Input is better this way.

xirzec commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,293 @@
+// @public
+export class LogsClient {
+    constructor(tokenCredential: TokenCredential, options?: LogsClientOptions);
+    // (undocumented)
```
ref docs

richardpark-msft replied:
Was leaving these unfilled in on purpose so I can make sure I don't forget to fill them in with actual content. 

Will be happening in the next few PRs.

richardpark-msft replied:
(also captured in my favorite issue: https://github.com/Azure/azure-sdk-for-js/issues/15296

xirzec commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,293 @@
+
+// @public
+export class LogsClient {
+    constructor(tokenCredential: TokenCredential, options?: LogsClientOptions);
```
no endpoint needed?

richardpark-msft replied:
The endpoint is optional (there is a global one that can be overridden).

xirzec commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,293 @@
+// @public
+export interface MetricDefinition {
+    category?: string;
+    dimensions?: LocalizableString_2[];
```
can we avoid having a `_2` typename?

richardpark-msft replied:
We were just talking about this on the feature crew. I believe we're going to try to just type these as 'plain' string as the feature for localizing these strings isn't actually available.

CC: @srnagar

richardpark-msft replied:
Also captured here: https://github.com/Azure/azure-sdk-for-js/issues/15296

xirzec commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,293 @@
+    id?: string;
+    isDimensionRequired?: boolean;
+    metricAvailabilities?: MetricAvailability[];
+    // Warning: (ae-forgotten-export) The symbol "LocalizableString" needs to be exported by the entry point index.d.ts
```
forgot exports

richardpark-msft replied:
I believe these are happening because of the multi-swagger-file issue with metrics (I can only export one of the multiple LocalizableString interfaces).

We want to stop exposing LocalizableString anyways so I'll have to restructure these models, which should eliminate this error. This is coming in the next few PRs for this package.

Recorded this in https://github.com/Azure/azure-sdk-for-js/issues/15296

xirzec commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,293 @@
+
+// @public (undocumented)
+export class MetricsClient {
+    constructor(tokenCredential: TokenCredential, options?: MetricsClientOptions);
```
I would expect `resourceUri` would be part of the constructor instead of being a required param on all methods? Or am I greatly confused on how this endpoint works?

richardpark-msft replied:
We can see how we make this more clear - it's not the arm resource id for the metrics server, it's the resource id that you're querying metrics on (ie, it could be a service bus, text analytics cluster, etc..). So it should be a parameter, but this is an oddity we'll want to consider.

richardpark-msft replied:
Captured this concern here: https://github.com/Azure/azure-sdk-for-js/issues/15296

xirzec commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,293 @@
+
+// @public (undocumented)
+export interface MetricsClientOptions extends PipelineOptions {
+    $host?: string;
```
can we rename this? `$host` is a little finnicky to declare in JS literals

richardpark-msft replied:
Yeah, was thinking 'endpoint' would be more appropriate based on other clients.

richardpark-msft replied:
Renamed to 'endpoint'

xirzec commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,293 @@
+
+// @public
+export type QueryGetResponse = QueryResults & {
+    _response: coreHttp.HttpResponse & {
```
if you're going to ship against `core-http` you should pull off all the `_response` shapes to avoid a future breaking change

richardpark-msft replied:
I was actually able to remove the entire model.

xirzec commented on sdk/monitor/monitor-query/review/monitor-query.api.md:
```diff
@@ -0,0 +1,293 @@
+}
+
+// @public (undocumented)
+export interface QueryStatistics {
```
is it worth making this a generic? Or is it really open ended in a way that's unknowable?

richardpark-msft replied:
I think we're going to do some more discussion about this. Most of the other teams are just leaving it completely untyped, but I'd like to see if we can actually go the other direction and add some more fields. 

There might still be some unmapped, however, but we can cross that bridge in between this release and the next, I think.

richardpark-msft replied:
btw, I have this covered in my "pre preview" issue: https://github.com/Azure/azure-sdk-for-js/issues/15296

xirzec commented on sdk/monitor/monitor-query/src/internal/modelConverters.ts:
```diff
@@ -0,0 +1,194 @@
+    metricnamespace: queryMetricsOptions?.metricNamespace
+  };
+
+  delete obj["orderBy"];
```
would it make more sense to use destructuring to avoid needing to do this? 

```ts
const { orderBy, metricNames, ...rest } = queryMetricsOptions;
const obj = {
   ...rest,
   orderby: orderBy,
   metric: metricNames?.join(",")
}
```

and so on?

richardpark-msft replied:
Oh wow, yeah, that'd be great :) I'll move over to that, just didn't think of it!

richardpark-msft replied:
Done!

xirzec commented on sdk/monitor/monitor-query/src/logsClient.ts:
```diff
@@ -0,0 +1,99 @@
+    // host.
+    const serviceClientOptions = createPipelineFromOptions(options || {}, authPolicy);
+
+    // TODO: this is bit odd, but I don't see a proper way to "opt out" of passing in a credential here.
```
In your swagger MD file you should be able to put `add-credentials: false` to avoid this problem

richardpark-msft replied:
Yes, did and it's now gone :)

xirzec commented on sdk/monitor/monitor-query/src/metricsClient.ts:
```diff
@@ -0,0 +1,125 @@
+
+export interface MetricsClientOptions extends PipelineOptions {
+  /** server parameter */
+  $host?: string;
```
prefer `host` for the public shape, you can map it to `$host` internally

richardpark-msft replied:
Removed and consolidated so it's just `endpoint` for both the metric and log client.

xirzec commented on sdk/monitor/monitor-query/src/metricsClient.ts:
```diff
@@ -0,0 +1,125 @@
+
+    this._metricsClient = new GeneratedMetricsClient(
+      tokenCredential,
+      MetricsApiVersion.TwoThousandSeventeen0501Preview,
```
We really need to fix this enum name because it is 🍌 🍌 🍌

richardpark-msft replied:
It's totally generated code. I'm a bit insulted you think I'd create an enum like that! (checks code).

xirzec replied:
Oh I know it's generated, my remark was about how we should fix the codegen, not that you did anything wrong. :)

xirzec commented on sdk/monitor/monitor-query/swagger/README.md:
```diff
@@ -0,0 +1,75 @@
+package-name: "monitor-log-query"
+clear-output-folder: true
+generate-metadata: false
+add-credentials: true
```
flip this to false for all clients

richardpark-msft replied:
flipped! Much nicer!

xirzec reviewed (changes requested):
This is looking much better! Thank you so much for all your work!!

Just a few last things I'd like to address in the public surface (or at least talk about) before we land this thing and also a few minor remarks in implementation

richardpark-msft commented:
> 
> 
> No concerns from me. The API View isn't approved yet, I'd like to hear more about workspace vs workspaceId vs workspaces, but I'm ok if that gets punted to another PR.

Yeah, I think moving that to the next PR makes sense. I have some feedback from the feature crew that I'll be merging in and this is one area I'd like to take on then.

xirzec reviewed (approved):
Ship it!  🚀
